12_glob_lab/globlab.txt

```diff
@@ -1,29 +1,30 @@
-Glob Lab.
-https://www.tocode.co.il/bundles/unix1/lessons/globlab
-
-1. Type in a glob pattern to match:
-
-  1. All files whose name starts with a lowercase letter
-  Answer: 
-
-  2. All files whose name starts with an uppercase letter
-  Answer:
-
-  3. All files whose name ends in an "a", "e", "i", "o" or "u"
-  Answer:
-
-  4. All files whose name DOES NOT end in an "a", "e", "i", "o" or "u"
-  Answer:
-
-  5. All files with extension ".c" that are in a folder named "src"
-  Answer:
-
-  6. All files with extension ".c" under any folder
-  Answer:
-
-2. Write a command that would delete all files ending with a ".png" under any subdirectory.
-
-3. The command "ls -l" can take the name of a file or directory and show details on them. From within "wildcards" folder run "ls -l [A-Z]*". What was printed? Why?
-
-4. Search in "man ls" for an option to show only directory data so the command "ls _____ D*" should print only the word Doc
-
+Glob Lab.
+https://www.tocode.co.il/bundles/unix1/lessons/globlab
+
+1. Type in a glob pattern to match:
+
+  1. All files whose name starts with a lowercase letter
+  Answer: ls [a-z]* or ls [[:lower:]]
+
+  2. All files whose name starts with an uppercase letter
+  Answer: ls [A-Z]*????
+       or ls [[:upper:]]
+
+  3. All files whose name ends in an "a", "e", "i", "o" or "u"
+  Answer:ls *[aeiou]
+
+  4. All files whose name DOES NOT end in an "a", "e", "i", "o" or "u"
+  Answer:ls *[!aeiou]
+
+  5. All files with extension ".c" that are in a folder named "src"
+  Answer:ls *.c /src

```

Comment:
This doesn't work. You called ls with the list of `.c` files in current directory AND the word `/src`. What it'll do is show all `.c` files in current directory and contents of `/src` folder if exists (but probably an error message since it doesn't usually exists)

What you want to do here is combine the two to a single glob pattern stating all files in src folder that ends in .c. The command is therefore: `ls src/*.c`

Note also that `/src` and `src` refer to 2 different directories: The first is at the top of the filesystem tree, and the second is relative to your current working directory. So if you're in `$HOME/lab` the second is actually `/home/moshiko/lab/src`, while the first is always just `/src` no matter from where you use it